tests/test_execution_compute_plan.py

```diff
@@ -150,46 +143,170 @@ def test_compute_plan_single_session_failure(global_execution_env):
+    spec = factory.create_aggregate_algo()
+    aggregate_algo = sessions[0].add_aggregate_algo(spec)
+
+    # launch execution
+    previous_aggregatetuple_spec = None
+    previous_composite_traintuple_specs = []
+
+    cp_spec = factory.create_compute_plan(objective=objective)
+
+    for round_ in range(number_of_rounds):
+        # create composite traintuple on each node
+        composite_traintuple_specs = []
+        for index, dataset in enumerate(datasets):
+            kwargs = {}
+            if previous_aggregatetuple_spec:
+                kwargs = {
+                    'in_head_model_tuple': previous_composite_traintuple_specs[index],
+                    'in_trunk_model_tuple': previous_aggregatetuple_spec,
+                }
+            spec = cp_spec.add_composite_traintuple(
+                composite_algo=composite_algo,
+                dataset=dataset,
+                data_samples=[dataset.train_data_sample_keys[0 + round_]],
+                **kwargs,
+            )
+            composite_traintuple_specs.append(spec)
+
+        # create aggregate on its node
+        spec = cp_spec.add_aggregatetuple(
+            aggregate_algo=aggregate_algo,
+            worker=aggregate_worker,
+            in_models_tuples=composite_traintuple_specs,
+        )
+
+        # save state of round
+        previous_aggregatetuple_spec = spec
+        previous_composite_traintuple_specs = composite_traintuple_specs
+
+    # last round: create associated testtuple
+    for composite_traintuple_spec in previous_composite_traintuple_specs:
+        cp_spec.add_testtuple(
+            traintuple_spec=composite_traintuple_spec,
+        )
+
+    cp = session_1.add_compute_plan(cp_spec).future().wait()
+    tuples = (cp.list_traintuple(session_1) +
+              cp.list_composite_traintuples(session_1) +
+              cp.list_aggregate_tuples(session_1) +
+              cp.list_testtuples(session_1))
+    for t in tuples:
+        assert t.status == 'done'
+
+
+def test_compute_plan_circular_dependency_failure(factory, session):
+    spec = factory.create_dataset()
+    dataset = session.add_dataset(spec)
+
+    spec = factory.create_algo()
+    algo = session.add_algo(spec)
+
+    spec = factory.create_data_sample(test_only=False, datasets=[dataset])
+    data_sample = session.add_data_sample(spec)
+
+    spec = factory.create_objective(dataset=dataset)
+    objective = session.add_objective(spec)
+
+    cp_spec = factory.create_compute_plan(objective=objective)
+
+    traintuple_spec_1 = cp_spec.add_traintuple(
+        dataset=dataset,
+        algo=algo,
+        data_samples=[data_sample]
+    )
+
+    traintuple_spec_2 = cp_spec.add_traintuple(
+        dataset=dataset,
+        algo=algo,
+        data_samples=[data_sample]
+    )
+
+    traintuple_spec_1.in_models_ids.append(traintuple_spec_2.id)
+    traintuple_spec_2.in_models_ids.append(traintuple_spec_1.id)
+
+    # TODO make sur the creation is rejected
```

Comment:
comment to be removed?
the `substra` library raising an `InvalidRequest` should be sufficient to ensure that the create has been rejected.

Reply:
Yes, it should have been removed when the pytest.raises was added.